hibernate-core/src/main/java/org/hibernate/dialect/function/StringFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.dialect.function;
+
+import org.hibernate.QueryException;
+import org.hibernate.metamodel.mapping.JdbcMapping;
+import org.hibernate.metamodel.model.domain.ReturnableType;
+import org.hibernate.query.sqm.function.AbstractSqmSelfRenderingFunctionDescriptor;
+import org.hibernate.query.sqm.produce.function.ArgumentTypesValidator;
+import org.hibernate.query.sqm.produce.function.StandardFunctionReturnTypeResolvers;
+import org.hibernate.sql.ast.SqlAstTranslator;
+import org.hibernate.sql.ast.spi.SqlAppender;
+import org.hibernate.sql.ast.tree.SqlAstNode;
+import org.hibernate.sql.ast.tree.expression.Expression;
+import org.hibernate.type.SqlTypes;
+import org.hibernate.type.StandardBasicTypes;
+import org.hibernate.type.descriptor.java.EnumJavaType;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+import org.hibernate.type.spi.TypeConfiguration;
+
+import java.util.List;
+
+import static org.hibernate.query.sqm.produce.function.FunctionParameterType.ENUM;
+
+
+/**
+ * The HQL {@code ordinal()} function returns the ordinal value of an enum
+ * <p>
+ * For enum fields mapped as ORDINAL it's a synonym for {@code cast(x as Integer)}. Same as {@link CastStrEmulation} but for Integer.
+ * For enum fields mapped as STRING or ENUM it's a case statement that returns the ordinal value.
+ *
+ * @author Luca Molteni
+ */
```

Comment:
Could you fix this javadoc please?

Reply:
Of course. Please check and if you have no suggestion, I can squash commits.